cmd/opampsupervisor/supervisor/supervisor_test.go

```diff
@@ -1557,3 +1540,70 @@ capabilities:
 		require.NoError(t, os.RemoveAll(tmpDir))
 	})
 }
+
+func TestSupervisor_exportLogsWithSDK(t *testing.T) {
+	template := `
+server:
+  endpoint: ws://localhost/v1/opamp
+  tls:
+    insecure: true
+
+capabilities:
+  reports_effective_config: true
+  reports_own_metrics: true
+  reports_health: true
+  accepts_remote_config: true
+  reports_remote_config: true
+  accepts_restart_command: true
+
+storage:
+  directory: %s
+
+agent:
+  executable: %s
+
+telemetry:
+  logs:
+    level: info
+    processors:
+      - batch:
+          exporter:
+            otlp:
+              protocol: http/protobuf
+              endpoint: http://127.0.0.1:4318
+`
+
+	cfg := setupSupervisorConfig(t, template)
+	supervisor, err := NewSupervisor(zap.NewNop(), cfg)
+	require.NoError(t, err)
+
+	path := filepath.Join(t.TempDir(), "output.txt")
+	backend := testbed.NewOTLPHTTPDataReceiver(4318)
+	mockBackend := testbed.NewMockBackend(path, backend)
+	err = mockBackend.Start()
+	require.NoError(t, err)
+
+	mockBackend.EnableRecording()
+	supervisor.telemetrySettings.Logger.Info("test log")
+	time.Sleep(4 * time.Second)
+	mockBackend.Stop()
+
+	receivedLogs := mockBackend.ReceivedLogs
+	require.Len(t, receivedLogs, 1)
+	l := mockBackend.ReceivedLogs[0]
+	require.Equal(t, 1, l.ResourceLogs().Len())
```

Comment:
Could we put this in a `require.Eventually` so we can get rid of the sleep above?

Reply:
Thats a cool feature, updated.

Reply:
What about it not being thread-safe causes problems here? With `require.Eventually` it should just poll the variable every so often; I don't think the order of operations really matters since we're not writing to it and don't care about a particular value for the length.